app/mikane/src/app/pages/expenditures/expense/expense.component.ts

```diff
@@ -0,0 +1,179 @@
+import { CommonModule } from '@angular/common';
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { MatButtonModule } from '@angular/material/button';
+import { MatCardModule } from '@angular/material/card';
+import { MatDialog, MatDialogModule } from '@angular/material/dialog';
+import { MatIconModule } from '@angular/material/icon';
+import { MatToolbarModule } from '@angular/material/toolbar';
+import { ActivatedRoute, Router, RouterLink } from '@angular/router';
+import { Subject, Subscription, filter, switchMap, takeUntil } from 'rxjs';
+import { MenuComponent } from 'src/app/features/menu/menu.component';
+import { Category, CategoryService } from 'src/app/services/category/category.service';
+import { ConfirmDialogComponent } from 'src/app/features/confirm-dialog/confirm-dialog.component';
+import { ExpenditureDialogComponent } from '../expenditure-dialog/expenditure-dialog.component';
+import { BreakpointService } from 'src/app/services/breakpoint/breakpoint.service';
+import { AuthService } from 'src/app/services/auth/auth.service';
+import { Expense, ExpenseService } from 'src/app/services/expense/expense.service';
+import { EventService, PuddingEvent } from 'src/app/services/event/event.service';
+import { MessageService } from 'src/app/services/message/message.service';
+import { ProgressSpinnerComponent } from 'src/app/shared/progress-spinner/progress-spinner.component';
+import { ApiError } from 'src/app/types/apiError.type';
+
+@Component({
+	templateUrl: 'expense.component.html',
+	styleUrls: ['./expense.component.scss'],
+	standalone: true,
+	imports: [
+		CommonModule,
+		MatCardModule,
+		ProgressSpinnerComponent,
+		MenuComponent,
+		MatDialogModule,
+		MatIconModule,
+		RouterLink,
+		MatButtonModule,
+		MatToolbarModule,
+	],
+})
+export class ExpenseComponent implements OnInit, OnDestroy {
+	protected loading: boolean = true;
+	expense: Expense;
+	event: PuddingEvent;
+	currentUserId: string;
+
+	private expenseSubscription: Subscription;
+	private authSubscription: Subscription;
+	cancel$: Subject<void> = new Subject();
+	destroy$: Subject<void> = new Subject();
```

Comment:
You can use this for all the subscriptions if you want. Instead of expenseSubscription and authSubscription